lib/rmt/cli/systems.rb

```diff
@@ -28,4 +31,39 @@ def scc_sync
     RMT::SCC.new(options).sync_systems
   end
 
+  desc 'remove TARGET', _('Permanently removes system with all its subscriptions and products.')
+  long_desc <<~REMOVE
+    #{_('Permanently removes selected system by login with all its subscriptions and products by login.')}
+
+    #{_('Examples')}:
+
+    $ rmt-cli systems remove uniqueLogin
+  REMOVE
+  def remove(target)
+    purge_system(target)
+  end
+
+  protected
+
+  def purge_system(target)
+    system = find_system(target)
+    destroy_system!(system)
+    puts _('Successfully removed system with login %{login}') % { login: target }
+  rescue SystemNotFoundException, SystemNotDestroyedException => e
+    puts e.message
+  end
+
+  private
+
+  def find_system(target)
+    System.find_by!(login: target)
+  rescue ActiveRecord::RecordNotFound
+    raise SystemNotFoundException.new(_('System with login %{login} not found.') % { login: target })
+  end
+
+  def destroy_system!(system)
+    system.destroy!
+  rescue ActiveRecord::RecordNotDestroyed
+    raise SystemNotDestroyedException.new(_('System with login %{login} cannot be removed') % { login: target })
```

Comment:
I'm not entirely sure how to test this one. @ikapelyukhin could you help me, please?

Reply:
> I'm not entirely sure how to test this one. @ikapelyukhin could you help me, please?

Sure, I'll take a look at this soon.

Reply:
@ikapelyukhin thank you so much for your time! I agree with your position about `system.destroy!`, but I also think that you might want to implement soft deletion later, and, in that case, having one method that can call `purge` or mark for deletion depending on a case would be nice. However, if that's not the case, I will make the necessary changes as your approach is much cleaner.

Reply:
> you might want to implement soft deletion later, and, in that case, having one method that can call `purge` or mark for deletion depending on a case would be nice.

In this case whoever implements this will make necessary refactoring and write the code that makes sense for that specific scenario there and then. You are sacrificing clean and maintainable code for solving a hypothetical problem, which maybe doesn't even need to be solved.

Reply:
you're right, thank you :+1: I will make the necessary changes today.